web/src/message_util.ts

```diff
@@ -112,3 +116,41 @@ export function get_topics_for_message_ids(message_ids: number[]): Map<string, [
     }
     return topics;
 }
+
+export function can_previous_pms_exist(recipient_ids_string: string): {
+    allow_local_echo: boolean;
+    can_previous_pms_exist: boolean;
+} {
+    // Check if there are any previous messages in the DM conversation.
+    const previous_direct_messages_exist = recent.has_conversation(recipient_ids_string);
+    if (previous_direct_messages_exist) {
+        return {allow_local_echo: true, can_previous_pms_exist: true};
+    }
+    // If not, we need to check if the current filter matches the DM view we
+    // are composing to.
+    const dm_conversation = current?.data?.filter.operands("dm")[0];
+    if (dm_conversation) {
+        const current_user_ids_string = people.emails_strings_to_user_ids_string(dm_conversation);
+        assert(current_user_ids_string !== undefined);
+        // If it matches, then there are certainly no messages in the conversation.
```

Comment:
I don't think this reasoning is correct. It's helpful to write comments like a proof. "If it matches, then there can be no messages in the conversation, because any non-empty current view will have already been in the pm_conversations data set queried above..."

That's helpful because it explains why you think something is true, and a reviewer can try to check that reasoning. In this case, I think that reasoning might be wrong? We need to also check `message_lists.current.fetch_status.has_found_newest` if we want to rely on the information; otherwise, this logic will be wrong if you click into an old DM conversation and the client is still fetching.

Reply:
Added an extra check for `message_lists.current.data.fetch_status.has_found_newest()`.